lib/services/database.go

```diff
@@ -527,6 +527,8 @@ func NewDatabaseFromRDSInstance(instance *rds.DBInstance) (types.Database, error
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
+	metadata.AssumeRoleARN = role.RoleARN
+	metadata.ExternalID = role.ExternalID
```

Comment:
alternatively `types.Database` can provide extra functions for setting and getting these two values to and from the spec metadata. then fetchers can overwrite these in a single place.

Reply:
I like that idea. I originally wrote this to have the fetchers call SetStatusAWS but I wanted to have the values in the spec as well